invenio_app_rdm/upgrade_scripts/migrate_8_0_to_9_0.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022 TU Wien.
+#
+# Invenio-App-RDM is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Record migration script from InvenioRDM 8.0 to 9.0.
+
+Disclaimer: This script is intended to be executed *only once*, namely when
+upgrading from InvenioRDM 8.0 to 9.0!
+If this script is executed at any other time, probably the best case scenario
+is that nothing happens!
+"""
+
+from invenio_rdm_records.records.api import RDMRecord
+from click import echo, secho
+from invenio_db import db
+
+
+def execute_upgrade():
+    """Execute the upgrade from InvenioRDM 8.0 to 9.0.
+
+    Please read the disclaimer on this module before thinking about executing
+    this function!
+    """
+
+    def update_funding_field(record):
+        try:
+            echo(f"Updating record: {record['id']}... ", nl=False)
+            for funding in record.metadata.get('funding', []):
+                award = funding.get('award', {})
+                funder = funding.get('funder', {})
+                if award.get('identifier') and award.get('scheme'):
+                    award_identifier = award.pop('identifier')
+                    award_scheme = award.pop('scheme')
+                    funding['award']['identifiers'] = [{
+                        'identifier': award_identifier,
+                        'scheme': award_scheme
+                    }]
+                if funder.get('identifier') and funder.get('scheme'):
+                    funder.pop('identifier')
+                    funder.pop('scheme')
+                if award.get('title', ''):
+                    award_title = award.pop('title')
+                    funding['award']['title'] = {'en': award_title}
+                if funder.get('title', ''):
+                    funder.pop('title')
```

Comment:
I think `title` was actually `name` (for "custom" funders). I'm not sure though if this is handled in the deposit form properly...